src/pynxtools/testing/nomad_example.py

```diff
@@ -0,0 +1,129 @@
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Test for NOMAD examples in reader plugins."""
+
+import os
+from typing import Any, Dict, List
+import tempfile
+import pytest
+
+try:
+    from nomad.config.models.plugins import (
+        ExampleUploadEntryPoint,
+    )
+    from nomad.datamodel import Context, EntryArchive
+    from nomad.parsing.parser import ArchiveParser
+except ImportError:
+    pytest.skip(
+        "Skipping NOMAD example tests because nomad is not installed",
+        allow_module_level=True,
+    )
+
+
+def get_file_parameter(example_path: str):
+    """
+    Get all example files for the plugin.
+
+    This function searches for specific example files in the given directory path.
+
+    Args:
+        example_path (str): Path to the directory containing example files.
+
+    Yields:
+        pytest.param: A pytest parameter object with the file path and file ID.
+    """
+    example_files = (
+        "schema.archive.yaml",
+        "schema.archive.yml",
+        "scheme.archive.yaml",
+        "scheme.archive.yml",
+        "schema.archive.json",
+        "scheme.archive.json",
+        "intra-entry.archive.json",
+    )
+
+    # Check if the provided path exists
+    if not os.path.exists(example_path):
+        raise FileNotFoundError(f"The directory '{example_path}' does not exist.")
+
+    # Walk through the specified directory
+    for root, _, files in os.walk(example_path):
+        for file in files:
+            normalized_file = file.lower()  # Normalize to lower case
+            if os.path.basename(normalized_file).endswith(example_files):
+                yield pytest.param(os.path.join(root, file), id=file)
```

Comment:
As far as I can remember, I had issue with the keyoward `yield` while running test in `pynxtools-stm`. Would you check it if there was something that I missed? Other than that it is fine.

Reply:
I had to change the structure of the pytest params in `pynxtools-stm` a bit (basically converting to a list, see https://github.com/FAIRmat-NFDI/pynxtools-stm/blob/7ee43248ee4cf74b5032138c21016cdf6b454f77/tests/test_nomad_examples.py#L52), but then it works.

Reply:
The only reason why you need to use list in pynxxtools-stm is that you want to concatenate the two generator outputs of this function because you run the same tests using two different folders. Actualy, it may even be better if you split the test [here](https://github.com/FAIRmat-NFDI/pynxtools-stm/blob/57d96d25efdef84b37fb945364ef17470bb6c36a/tests/test_nomad_examples.py#L44) into two different tests. In none of the other repos we have to use the list conversion, so I think it's fine with the `yield` keyword.

By the way, the reason why I am using this is that NOMAD uses the same functionality: https://gitlab.mpcdf.mpg.de/nomad-lab/nomad-FAIR/-/blob/develop/tests/parsing/test_archive_parser.py#L57